code/modules/atmospherics/components/binary_devices/circulator.dm

```diff
@@ -72,10 +82,35 @@
 /obj/machinery/atmospherics/binary/circulator/process()
 	..()
 
+	if(anchored && !(stat&BROKEN))
+		if ((last_pressure_delta > (5*ONE_ATMOSPHERE)) && (recent_moles_transferred > 0))
+			busy = TRUE
+		else
+			busy = FALSE
+
 	if(last_worldtime_transfer < world.time - 50)
 		recent_moles_transferred = 0
 		update_icon()
 
+	//If the gas exceeds temperature rating, the circulator leaks the gas and explodes.
+	if(air1.temperature > max_temperature)
```

Comment:
This assumes that `air1` always exists, are you sure this is the case? Otherwise, it would runtime